### DISCUSSION TITLE
Sending E-Mails from an iCloud Custom Domain, in case anyone else wants to

### DISCUSSION THREAD

**Discussion by OddMagnet** [goauthentik/authentik#11863] [Show and tell]
Hey! 

I was trying to send an email from my domain via iCloud custom domain and stumbled upon some problems.
Couldn't really find a solution by googling, so I thought I'd post it here so others can find it. 
Feel free to ask questions or add more information if you think it'd be helpful for others.

I used the `docker compose exec worker ak test_email my-name@gmail.com` command for testing
The values I used are from here: https://support.apple.com/en-us/102525
This is how the environment variables in my compose file look (attached to the worker container):
```
      # E-Mail Config
      # SMTP Host Emails are sent to
      - AUTHENTIK_EMAIL__PORT=587
      - AUTHENTIK_EMAIL__HOST=smtp.mail.me.com
      # Optionally authenticate (don't add quotation marks to your password)
      - AUTHENTIK_EMAIL__USERNAME=authentik@my-domain.com
      - AUTHENTIK_EMAIL__PASSWORD=app-specific-password-generated-in-icloud
      # Use StartTLS
      - AUTHENTIK_EMAIL__USE_TLS=true
      # Use SSL
      - AUTHENTIK_EMAIL__USE_SSL=false
      - AUTHENTIK_EMAIL__TIMEOUT=10
      # Email address authentik will send from, should have a correct @domain
      - AUTHENTIK_EMAIL__FROM=authentik@my-domain.com
``` 

However, here I got the following error:
`smtplib.SMTPAuthenticationError: (535, b'5.7.8 Error: authentication failed')`
The app specific password was correct, since I copied and pasted it.

I then tried using my apple-id for the `AUTHENTIK_EMAIL__USERNAME`, but then I get this error:
`smtplib.SMTPAuthenticationError: (550, b'5.1.1 Mailbox does not exist')`

**The Solution**
Since my Apple-ID was an `@googlemail.com` address neither the full address (my-name@googlemail.com) nor just the name (my-name) worked.
I remembered that I had 2 aliases for my primary mail:
`first-name.last-name@icloud.com` and `first-name.last-name@me.com`
Simply using `first-name.last-name` for `AUTHENTIK_EMAIL__USERNAME` worked for me.

---

Hope this helps someone else in the future :)

**Comment by mikechase3** [comment]
Boy that was a b\*\*\*\*. I did a write-up [here](https://notebook.mchase.me/career/compsci-and-development/domain-administration/how-to-setup-icloud-mail-server-settings-for-custom-domains) if you still need help.

Once you have a custom email/domain your "icloud" one becomes hidden in the iCloud mail web app. To find it, go to your account settings and look for manage your email & phone number. That'll list your old, original iCloud one.

<img width="1778" height="1290" alt="Brave Browser 2025-10-18 07 52 19" src="https://github.com/user-attachments/assets/ad1efa4b-f6f8-4984-90fd-0bc9f6f3ad59" />

TLDR: it's not your Apple ID's primary email, it's not your custom domain's email, **your username is your icloud.com email** you never use.

**Reply by alicancakil** [reply]
thank you very much @mikechase3

I was struggling with this as well 🙏
